src/blocks/Share/Share.tsx

```diff
@@ -30,6 +31,10 @@ const b = block('share-block');
 
 const Share = ({items, title}: ShareBlockProps) => {
     const {pathname, hostname} = useContext(LocationContext);
+    const defaultEvent = useAnalyticsEvent(DefaultEventNames.ShareButton);
+    const handleAnalytics = useAnalytics(defaultEvent);
+
+    const handleButtonClick = useCallback(() => handleAnalytics(), [handleAnalytics]);
```

Comment:
think we can just return memoized callback from `useAnalytics` not to memoize it every time

Reply:
I am not sure, it is necessary. We use such a callback only twice: in a Share block and In a Quote block. Other blocks use handleAnalytics inside their more complex callbacks. They will need their own useCallback wrapper anyway

On the other hand handleAnalytics can receive events as a props. And this (event's) type is not siutable for onClick props. So we need to handle this circumstance some way. I think it is too much for this case)